core/src/main/scala/cats/MonoidK.scala

```diff
@@ -38,3 +47,13 @@ import simulacrum.typeclass
       def combine(x: F[A], y: F[A]): F[A] = self.combine(x, y)
     }
 }
+
+trait CompositeMonoidK[F[_],G[_]]
+  extends MonoidK[λ[α => F[G[α]]]] {
+
+  implicit def F: MonoidK[F]
+  implicit def G: MonoidK[G]
```

Comment:
What I meant is that there is usage of the form `f compose g` for this entire family. I think we should preserve that, and also add an operator of the form `f.composedWith[G]` for the cases where the constraint isn't necessary.